boards/arm/beagle_bcf/beagleconnect_freedom.dts

```diff
@@ -127,16 +118,25 @@
 		reg = <0x4>;
 	};
 
-	light: opt3001-light@44 {
+	sens_i2c: sensor-switch {
```

Comment:
This node is missing a `@N` peripheral address.

Reply:
Hi, is the peripheral address supposed to signify something or is it supposed to just unique for same types of nodes?

Reply:
@cfriedt I think I somewhat understand, i2c senors should be defined as children of i2c controller, and the analog switch should be either child of gpio node or be in its original position.
However, I am still not sure how to show that the gpio switch drives the i2c sensors without making sensors as child nodes to the switch. Is there some primitive in device tree or maybe a new property needs to be added to gpio i2c switch bindings.

Reply:
> However, I am still not sure how to show that the gpio switch drives the i2c sensors

Unless there is some dedicated binding for a chip-enable pin (which isn't part of the i2c spec, but may be present on certain sensor bindings), and unless that pin is wired in to the same GPIO in this case, I'm not sure if it's possible to have software (e.g. i2c controller driver, hypothetically) automatically enable it.

It's really easy to have application code set or clear the gpio though - and there is minimal Devicetree involved. In the case that you still want a binding for the gpio i2c switch, you could give it a phandle to the gpio controller and an index representing the offset of the particular driver for the pin.

It's kind of weird that this node is an i2c controller though.

I would just have application code get a phandle to the gpio device and flip it as necessary.

Reply:
I think that is prone to error and potential race conditions. By defining a bus, we avoid mistakes and enable even shell tools to see the sensor i2c bus properly. Originally, I was looking at this as a bridge, but I can see that by not having an I2C address, it does not look like a peripheral to the primary I2C bus controller. To that end, I think we could use a different property to define the sensor I2C bus controller, rather than make it look like a peripheral. Does that make sense?